### PR TITLE
Fix Issue #304 Support setting namespaces for operations

### PR DESCRIPTION
**Pull request by VikingsFan**

### CODE REVIEW

File: OData/src/System.Web.OData/OData/Builder/ProcedureConfiguration.cs [hunk 1]
```diff
@@ -73,10 +74,15 @@ internal ProcedureConfiguration(ODataModelBuilder builder, string name)
...
         public string FullyQualifiedName
         {
-            get { return ModelBuilder.Namespace + "." + Name; }
+            get { return Namespace + "." + Name; }
         }
 
         /// <summary>
+        /// The Namespace by default is the ModelBuilder's Namespace.
+        /// </summary>
+        public string Namespace { get; set; }
```
Comment (xuzhg) [inline OData/src/System.Web.OData/OData/Builder/ProcedureConfiguration.cs]: how about if customer sets the namespace as empty or null?

**Comment by xuzhg** [comment]
Others look good to me. Thanks.

**Comment by VikingsFan** [comment]
Checked in 483ce56fa63b6950fd54f8a9119f866c2164468e

File: OData/src/System.Web.OData/OData/Builder/ProcedureConfiguration.cs [hunk 2]
```diff
@@ -25,6 +26,7 @@ internal ProcedureConfiguration(ODataModelBuilder builder, string name)
         {
             Name = name;
             ModelBuilder = builder;
+            _namespace = null;
```
Comment (congysu) [inline OData/src/System.Web.OData/OData/Builder/ProcedureConfiguration.cs]: Remove this line. `_namespace` initializes to `null` by default.

File: OData/src/System.Web.OData/OData/Builder/ProcedureConfiguration.cs [hunk 3]
```diff
@@ -73,7 +75,16 @@ internal ProcedureConfiguration(ODataModelBuilder builder, string name)
...
         public string FullyQualifiedName
         {
-            get { return ModelBuilder.Namespace + "." + Name; }
+            get { return Namespace + "." + Name; }
+        }
+
+        /// <summary>
+        /// The Namespace by default is the ModelBuilder's Namespace.
+        /// </summary>
+        public string Namespace
```
Comment (congysu) [inline OData/src/System.Web.OData/OData/Builder/ProcedureConfiguration.cs]: Add this to public API review.

**Comment by congysu** [comment]
@vojtechvit @brjohnstmsft Can you help review the change?  Thanks.

File: OData/test/UnitTest/System.Web.OData.Test/OData/MetadataControllerTest.cs [hunk 4]
```diff
@@ -917,6 +917,21 @@ public void ServiceDocument_OnlyContainOneFunctonImport_ForOverloadFunctions()
...
+        public void ServiceDocument_FunctionNamesapce_Configuration()
+        {
+            // Arrange
+            HttpServer server = new HttpServer(GetConfiguration());
+            HttpClient client = new HttpClient(server);
+
+            // Act
+            var response = client.GetAsync("http://localhost/$metadata").Result;
+            var responseString = response.Content.ReadAsStringAsync().Result;
+
```
Comment (congysu) [inline OData/test/UnitTest/System.Web.OData.Test/OData/MetadataControllerTest.cs]: `// Assert`

File: OData/test/UnitTest/System.Web.OData.Test/OData/ODataActionParametersTest.cs [hunk 5]
```diff
@@ -75,6 +75,25 @@ public void Can_find_action_overload_using_bindingparameter_type(string url)
             Assert.Equal("Wash", action.Name);
         }
 
+        [Theory]
+        [InlineData("Vehicles(Model=8,Name='8')/System.Web.OData.Builder.TestModels.Car/customize.NSAction")]
+        [InlineData("MyVehicle/System.Web.OData.Builder.TestModels.Car/customize.NSAction")]
+        public void Can_find_customized_namespace_action(string url)
```
Comment (congysu) [inline OData/test/UnitTest/System.Web.OData.Test/OData/ODataActionParametersTest.cs]: Capitalize Each Word.

**Comment by vojtechvit** [comment]
looks good!

File: OData/test/UnitTest/System.Web.OData.Test/OData/MetadataControllerTest.cs [hunk 6]
```diff
@@ -917,6 +917,21 @@ public void ServiceDocument_OnlyContainOneFunctonImport_ForOverloadFunctions()
         }
 
         [Fact]
+        public void ServiceDocument_FunctionNamesapce_Configuration()
```
Comment (brjohnstmsft) [inline OData/test/UnitTest/System.Web.OData.Test/OData/MetadataControllerTest.cs]: _FunctionNamespace_ (spelling)

**Comment by brjohnstmsft** [comment]
Changes look good to me. Thanks!

**Pull request closed without merging**